### PR TITLE
Consolidate dependencies into a single file which can be included by all

### PR DESCRIPTION
**Pull request by derekwbrown**

### CODE REVIEW

File: config/software/agent-deps.rb [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  dependency 'procps-ng'
+  dependency 'sysstat'
+end
+# Ship supervisor anywhere but on Windows
+if not windows?
+  dependency 'supervisor'
+  dependency 'zlib'
+else
+  # We use our own supervisor shipped as a py2exe-built executable on Windows...
+  # therefore we need py2exe. We also need psutil for our home-made supervisor.
```
Comment (degemer) [inline config/software/agent-deps.rb]: Nit: Could you delete the reference to psutil? It's a core dep now.

File: config/software/datadog-agent-integrations.rb [hunk 2]
```diff
@@ -93,11 +93,16 @@
...
 
   pip_cmd = "install --install-option=\"--install-scripts=#{windows_safe_path(install_dir)}/bin\" -c #{install_dir}/agent/requirements.txt -r /check_requirements.txt"
-  build_env = {
-    "LD_RUN_PATH" => "#{install_dir}/embedded/lib",
-    "PATH" => "/#{install_dir}/embedded/bin:#{ENV['PATH']}",
-  }
-  command "pip #{pip_cmd}", :env => build_env
+  if windows?
+    inst_cmd = "#{windows_safe_path(install_dir)}\\embedded\\scripts\\pip.exe " + pip_cmd
+    command inst_cmd
```
Comment (degemer) [inline config/software/datadog-agent-integrations.rb]: The `pip` DSL already does that - https://github.com/DataDog/omnibus-ruby/blob/datadog-5.5.0/lib/omnibus/builder.rb#L96-L103, so we can use it (we were using it previously, not sure why it disappeared)

File: config/software/datadog-agent-integrations.rb [hunk 3]
```diff
@@ -93,11 +93,16 @@
...
+  else
+    build_env = {
+      "LD_RUN_PATH" => "#{install_dir}/embedded/lib",
+      "PATH" => "/#{install_dir}/embedded/bin:#{ENV['PATH']}",
+    }
+    command "pip #{pip_cmd}", :env => build_env
```
Comment (degemer) [inline config/software/datadog-agent-integrations.rb]: Same, `pip` accepts options

File: config/software/agent-deps.rb (hunk 1 above)
Comment (derekwbrown) [inline config/software/agent-deps.rb]: I tried w/o psutil, and the build failed because it couldn't find psutil.  So I put it back in.

Comment (degemer) [inline config/software/agent-deps.rb]: Ah yes, I was talking of the comment :)

**Pull request merged**